### PR TITLE
Feature: API

### PR DESCRIPTION
**Pull request by 2004-okari**
- Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend
- Fetched my books from the Bookstore API after the component has rendered
- Refactored my add book and remove book features to persist your changes in the server
- My application behaves the same way after the refactoring

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -1,29 +1,84 @@
...
-
-// ...
+const API_ID = 'u3BFViY5zE36mpKmwqF4';
+const API_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
 
 function BookList() {
-  const books = useSelector((state) => state.books.books);
   const dispatch = useDispatch();
+  const library = useSelector((state) => state.books.books);
+  // const books = Object.values(library).reduce((acc, books) => acc.concat(books), []);
```
Comment (ademibowale) [inline src/components/BookList.js]: - [ ] Great job on implementing the functionality based on the project requirement but I will kindly suggest you remove the commented code line from your code base This makes your code look so messy and look distractive to another developer when reading your code please kindly fix it .

File: src/components/BookList.js [hunk 2]
```diff
@@ -1,29 +1,84 @@
...
+  }));
+
+  console.log(books);
+  const loading = useSelector((state) => state.books.loading);
+  const error = useSelector((state) => state.books.error);
+
+  // eslint-disable-next-line
+  const deleteResource = async (item_id) => {
+    try {
+      // eslint-disable-next-line
```
Comment (ademibowale) [inline src/components/BookList.js:25]: - [ ] You are currently disabling eslint in this file, kindly fix the linters rather than disabling eslint 👍

**Review by ademibowale — CHANGES_REQUESTED** [review]
## STATUS: Changes Required :recycle:

### Hi @2004-okari  , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image


### To Highlight 🥇

✅ Descriptive Pull Request title and summary. ✔️
✅ Good use of Github flow ✔️
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

⚠️⚠️ Important Note ⚠️⚠️
Please don't open the new PR, please make the above changes in this branch and submit this same PR https://github.com/NalianAchei2020/Hello_Microverse/pull/1 for getting another review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookList.js (hunk 2 above)
Comment (2004-okari) [reply src/components/BookList.js:25]: Due to the nature of how the API displays id, the property is item_id  from the Api , The linter requires me to use camelCase which is inevitable

**Review by mikethreels — APPROVED** [review]
Hi Nyandika, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good communication with the CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**